BUILD

```diff
@@ -0,0 +1,54 @@
+################################# define local and nolocal #################################
+# Conditional compile definitions based on --define=local=<value> build flag
+config_setting(
+    name = "is_local",
+    values = {"define": "local=true"},
+)
+
+# Set defines based on whether LOCAL is defined
+local_defines = select({
+    "//conditions:default": ["NOLOCAL=1"],  
+    ":is_local": ["LOCAL=1"],               
+})
+
+################################# library BUILD target (files in src and include) #################################
+cc_library(
+    name = "include_files",
+    hdrs = glob(["include/**/*.h"]),
+    deps = ["@com_github_nlohmann_json//:json"],
+    visibility = ["//visibility:public"],
+)
+
+cc_library(
+    name = "src_files",
+    srcs = glob(["src/**/*.cpp"]),
+    hdrs = glob(["src/**/*.h"]),
+    visibility = ["//visibility:public"],
+    deps = [":include_files"],
+    defines = local_defines, 
+)
+
+################################# main BUILD target #################################
+cc_binary(
+    name = "bootstrap",
+    srcs = ["main.cpp"],
+    copts = ["-std=c++20", "-Wall", "-pthread"],
```

Comment:
To avoid writing the same copts and linkopts every time for each cc_binary, you can put them in a `.bazelrc` file in the project root directory with the following content:
```
build --cxxopt=-std=c++20
build --copt=-Wall
build --copt=-pthread
build --linkopt=-lpthread
```

I tried this on my linux vm and it worked for me.

Reply:
done!